Doc/library/email.mime.rst

```diff
@@ -180,11 +180,11 @@ Here are the classes:
    A subclass of :class:`~email.mime.nonmultipart.MIMENonMultipart`, the
    :class:`MIMEImage` class is used to create MIME message objects of major type
    :mimetype:`image`. *_imagedata* is a string containing the raw image data.  If
-   this data can be decoded by the standard Python module :mod:`imghdr`, then the
-   subtype will be automatically included in the :mailheader:`Content-Type` header.
-   Otherwise you can explicitly specify the image subtype via the *_subtype*
-   argument.  If the minor type could not be guessed and *_subtype* was not given,
-   then :exc:`TypeError` is raised.
+   this data can be decoded by the standard Python module :mod:`imghdr`
+   (which is deprecated), then the subtype will be automatically included in the
```

Comment:
`imghdr.what()` is only 18 lines of code and the rest of `imghdr.py` is just tests and an `__main__`.  Why don't we just inline that into the email package and either expose it as a public API or just leave it private?

Reply:
If we did that how would you document that? Document all file formats in the constructor's docs?